research/engine/pegasus/network/src/manager.rs

```diff
@@ -58,7 +58,36 @@ impl ServerManager {
 }
 
 impl ServerDetect for Vec<Server> {
-    fn fetch(&mut self) -> &[Server] {
-        self.as_slice()
+    fn fetch(&self) -> Vec<Server> {
+        self.clone()
     }
 }
+
+pub struct SimpleServerDetector {
+    peers_mutex: Mutex<Vec<Server>>,
+}
+
+impl SimpleServerDetector {
+    pub fn new() -> Self {
+        SimpleServerDetector { peers_mutex: Mutex::new(vec![]) }
+    }
+
+    pub fn update_peer_view<Iter: Iterator<Item = (u64, SocketAddr)>>(&self, peer_view: Iter) {
+        let new_peers = peer_view.map(|(id, addr)| Server { id, addr }).collect::<Vec<Server>>();
+        let mut peers = self.peers_mutex.lock().expect("unexpected error locking when update peer view");
```

Comment:
Will this `expect()` cause any issue? I mean if it is used while running a service, then it should not be down due to such an error; if it is just for initiating a service, then it is fine.

Reply:
This is just for initiating the service (used when starting service with v2 store).